botorch/models/higher_order_gp.py

```diff
@@ -650,7 +654,7 @@ def posterior(
                 pred_variance = self.make_posterior_variances(joint_covar)
 
                 full_covar = KroneckerProductLazyTensor(
-                    full_covar, *joint_covar.lazy_tensors[1:]
+                    full_covar, *[x.detach() for x in joint_covar.lazy_tensors[1:]]
```

Comment:
I guess this will preclude computing gradients of quantities computed on the posterior w.r.t. the latents? I think that's fine since we usually don't do that, but we should probably add a comment here just to be safe.

Reply:
I think it's unlikely we'd need to do that for now, but the updated comment makes it clear.